pkg/daemon/daemon.go

```diff
@@ -254,14 +263,41 @@ func cmdRun(p *ptpProcess) {
 		return
 	}
 
+	if stdoutToSocket {
+		connected = false
+		for {
+			c, err = handleConnection()
+			if err != nil {
+				glog.Errorf("error trying to connect to event socket. retrying in %d secs", connectionRetryInterval)
+				time.Sleep(connectionRetryInterval)
+				continue
+			}
+			connected = true
+			break
+		}
+	}
+
 	done := make(chan struct{})
 
 	scanner := bufio.NewScanner(cmdReader)
 	go func() {
 		for scanner.Scan() {
 			output := scanner.Text()
-			fmt.Printf("%s\n", output)
-			extractMetrics(p.name, p.iface, output)
+			if stdoutToSocket {
+				out := fmt.Sprintf("%s\n", output)
```

Comment:
you need to move this outside of the if because you remove the /n from the else print